protocol/daemons/pricefeed/client/client.go

```diff
@@ -49,6 +50,33 @@ type Client struct {
 
 	// Ensure stop only executes one time.
 	stopDaemon sync.Once
+
+	// isHealthy indicates whether the daemon is healthy - i.e., it is actively running and calculating prices.
+	// The daemon is considered healthy when it is producing non-empty price updates. It will be considered unhealthy
+	// until it produces its first non-empty price update.
+	isHealthy atomic.Bool
+}
+
+// Ensure Client implements the DaemonClient interface.
+var _ daemontypes.DaemonClient = (*Client)(nil)
+
+// HealthCheck returns an error if the daemon is unhealthy.
+// The daemon is considered unhealthy if it is not producing non-empty price updates. We expect a short period of time
+// to pass on daemon startup where the daemon is unhealthy as the price cache warms up.
+// This method is synchronized by isHealthy, which is an atomic.Bool.
+func (c *Client) HealthCheck(_ context.Context) error {
+	if !c.isHealthy.Load() {
+		return errors.New(
```

Comment:
Consider using a `mutex` around an `error` so that you can provide details around the error.

This would allow you to differentiate between an error due to initialization failure or due to price update issues.